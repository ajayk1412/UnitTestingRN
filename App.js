import {Component} from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import ShowCounterValue from './ShowCounterValue';
import styles from './styles';
import Users from './Users';

export default class App extends Component {
  constructor(props) {
    super(props);
    console.log('Parent constructor');
    this.state = {
      counterValue: 0,
    };
  }

  componentDidMount() {
    // console.log('ParentcomponentDidMount');
    Users.all().then(data => console.log('check api data', data));
  }

  increment = x => {
    // return x * 10;
    this.setState({counterValue: x});
  };

  render() {
    console.log('Parent render');
    return (
      <View style={styles.mainContainer}>
        <TouchableOpacity
          testID="increment"
          style={styles.buttonStyle}
          onPress={this.increment}>
          <Text>Increment Value12</Text>
        </TouchableOpacity>
        <ShowCounterValue counterValue={this.state.counterValue} />
      </View>
    );
  }
}
