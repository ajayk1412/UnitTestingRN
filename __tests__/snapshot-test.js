import 'react-native';
import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

it('Snapshot testing', () => {
  const snap = renderer.create(<App />).toJSON();
  expect(snap).toMatchSnapshot();
});
