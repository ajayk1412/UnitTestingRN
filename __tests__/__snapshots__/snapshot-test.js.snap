// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "green",
        "borderRadius": 8,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="increment"
  >
    <Text>
      Increment Value
    </Text>
  </View>
  <View>
    <Text>
      0
    </Text>
  </View>
</View>
`;
